### ISSUE TITLE
GiMP thumbnails for .xcf files appear only in Gimp, not in file explorers.

### ISSUE THREAD

**Issue by Coeur-Noir** [snapcrafters/gimp#125]
Hi,

Gimp as a snap stores its thumbnails in `~/snap/gimp/common/.cache/thumbnails/`

It's not the regular place to store thumbnails, which is `~/.cache/thumbnails/`

By doing so, gimp-as-a-snap prevents other applications to make use of thumbnails for .xcf files, most notably files explorer ( Nemo, Nautilus and the like ).

Workaround for the moment, adding these lines to my fstab : 
```
# .xcf thumbnails from gimp snap for any app
/home/user_1/.cache/thumbnails/ /home/user_1/snap/gimp/common/.cache/thumbnails/    none    bind
/home/user_2/.cache/thumbnails/ /home/user_2/snap/gimp/common/.cache/thumbnails/    none    bind
/home/user_3/.cache/thumbnails/ /home/user_3/snap/gimp/common/.cache/thumbnails/    none    bind
```
Now I can see thumbnails for my .xcf files in Nemo and Nautilus.
I wrote « workaround for the moment » as I assume it won't hold long term as it kind of breaks some safety design.

**So to make it short here : the bug is, as soon as you use gimp as a snap, you lose the ability of thumbnails for .xcf files in files explorers**. Which is a painful regression when you work hours a day manipulating images.

Longer story :
⋅ https://forum.snapcraft.io/t/gimp-and-thumbnails-for-xcf-files/18131
⋅ https://forum.ubuntu-fr.org/viewtopic.php?id=2053949
⋅ https://discourse.ubuntubudgie.org/t/gimp-as-a-snap-no-thumbnails-for-xcf-files-in-nemo-gthumb/3799
⋅ https://bugs.launchpad.net/ubuntu/+source/gimp/+bug/1883457
⋅ https://askubuntu.com/questions/1250031/gimp-snap-and-thumbnails-for-xcf-files

**Comment by Coeur-Noir** [comment]
Note, also : there is no longer any thumbnailer for .xcf files available system side.
By system side I mean outside of gimp and outside of snap framework.

_gnome-xcf-thumbnailer_ is dead ( can't work with gimp 2.10.x ) so probably best solution might be to have a new updated thumbnailer for xcf files ?

**Comment by petaflot** [comment]
wouldn't this workaround be simpler?

```
# preserve already existing thumbnails
mv ~/snap/gimp/common/.cache/thumbnails/* ~/.cache/thumbnails/
# rmdir should work, but there may be hidden files
rm -rf ~/snap/gimp/common/.cache/thumbnails/
# now all thumbnails will be at the same place
ln -s  ~/.cache/thumbnails/ ~/snap/gimp/common/.cache/thumbnails
```

**Comment by petaflot** [comment]
also see 
- https://gitlab.xfce.org/xfce/tumbler/-/issues/64
- https://gitlab.gnome.org/GNOME/gimp/-/issues/8396

**Comment by diddledani** [comment]
No, @petaflot , that won't work at all and will potentially break GIMP in the process. The sandbox explicitly denies access to "dotfiles" in the user's home directory, including `.cache/`.

**Comment by Coeur-Noir** [comment]
A **mount bind** works, at least used to work, I gave up on Gimp as a snap, for that inability to show its thumbnails outside Gimp itself, amongst other things.

**Comment by Coeur-Noir** [comment]
@diddledani  The symbolic link will work if you replace the legacy folder `~/.cache/thumbnails` by a symlink that targets `~/snap/gimp/common/.cache/thumbnails` because in that way, Gimp does not need to go out of its sandbox since it's actually all other applications which will look into `~/snap/…` and I think it's what @petaflot's suggested.

It's the first « workaround » that came to my mind ago but I was worried all thumbnails for a session ended up stored into one particular « app » config folder - that's no sense. As previously said, I don't know if it still works with latest version of `snapd`

**Comment by jnsgruk** [comment]
Hi there! We recently merged a big overhaul of the GIMP snap, keeping us up to date with the latest 3.0 feature release.

The snap has been rebased on core24, has migrated to a much improved GNOME extension & content snap, and is now using many more components from the Ubuntu archive. Given either the version stated in this issue, or the last time the issue was updated, it's highly likely that the issue is resolved or is no longer relevant to the supported version of GIMP.

As part of getting the snap and the repo back on track, and to help us prioritise ongoing issues, I'm working through all issues that refer to older versions or haven't been updated in a while.

I'm closing this issue for now, but please comment if the issue still persists, and I'll reopen and prioritise getting it fixed, or at least get you an update!
